Let's solve conversions that are reported as errors.

--- a/src/VBox/HostDrivers/VBoxNetFlt/win/cfg/VBoxNetCfg.cpp	2020-04-09 19:41:59 +0000
+++ b/src/VBox/HostDrivers/VBoxNetFlt/win/cfg/VBoxNetCfg.cpp	2020-05-06 20:44:06 +0000
@@ -881,7 +883,7 @@

 VBOXNETCFGWIN_DECL(VOID) VBoxNetCfgWinSetLogging(IN LOG_ROUTINE pfnLog)
 {
-    *((void * volatile *)&g_Logger) = pfnLog;
+    *((void * volatile *)&g_Logger) = (void *)pfnLog;
 }

 /*
--- a/src/VBox/HostDrivers/VBoxUSB/win/install/USBInstall.cpp	2020-04-09 19:42:00.000000000 +0200
+++ b/src/VBox/HostDrivers/VBoxUSB/win/install/USBInstall.cpp	2020-05-16 19:02:06.095926900 +0200
@@ -117,7 +117,7 @@
         if (RT_SUCCESS(rc))
         {
             /* Install the INF file: */
-            HRESULT hr = VBoxDrvCfgInfInstall(pwszInfFile);
+            HRESULT hr = VBoxDrvCfgInfInstall((wchar_t*)pwszInfFile);
             if (hr == S_OK)
                 RTPrintf("Installation successful.\n");
             else
